milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
That is correct -- you would still have to handle for the exception -- @douglashall's point was that the validation step is unnecessary because of that fact, and removing this unnecessary check would eliminate an unnecessary database call.  So heck, let's do it! 😄